web/src/beta/features/ProjectSettings/innerPages/PublicSettings/AliasSettings/index.tsx

```diff
@@ -5,42 +5,55 @@ import {
 } from "@reearth/beta/features/Editor/Publish/PublishToolsPanel/common";
 import { Button, Icon } from "@reearth/beta/lib/reearth-ui";
 import CommonField from "@reearth/beta/ui/fields/CommonField";
+import {
+  useProjectFetcher,
+  useStorytellingFetcher
+} from "@reearth/services/api";
 import { config } from "@reearth/services/config";
 import { useT } from "@reearth/services/i18n";
 import { useNotification } from "@reearth/services/state";
 import { styled, useTheme } from "@reearth/services/theme";
 import { FC, useCallback, useMemo, useState } from "react";
 
 import { PublicAliasSettingsType } from "..";
+import {
+  SettingsProjectWithTypename,
+  StoryWithTypename
+} from "../PublicSettingsDetail";
 
 import EditPanel from "./EditPanel";
 
 export type AliasSettingProps = {
-  alias: string;
   isStory?: boolean;
-  defaultAlias?: string;
+  alias?: string;
+  settingsItem?: SettingsProjectWithTypename | StoryWithTypename;
   onUpdateAlias?: (settings: PublicAliasSettingsType) => void;
-  onClose?: () => void;
-  onSubmit?: (alias?: string) => void;
 };
+
 const AliasSetting: FC<AliasSettingProps> = ({
-  alias,
   isStory,
-  defaultAlias,
+  settingsItem,
   onUpdateAlias
 }) => {
   const theme = useTheme();
   const t = useT();
+  const { checkProjectAlias } = useProjectFetcher();
+  const { checkStoryAlias } = useStorytellingFetcher();
+
   const [, setNotification] = useNotification();
 
   const [open, setOpen] = useState(false);
   const handleOpen = useCallback(() => setOpen(true), []);
   const handleClose = useCallback(() => setOpen(false), []);
 
+  const alias = useMemo(
+    () => (settingsItem?.alias ? settingsItem.alias : settingsItem?.id),
+    [settingsItem?.alias, settingsItem?.id]
+  );
```

Comment:
Currently we don't use id as the default alias, i think we can remove this fallback and relies on `alias` field totally.